clients/src/test/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinatorTest.java

```diff
@@ -258,6 +259,45 @@ public void testJoinGroupRequestWithMemberIdMisMatch() {
         assertFalse(future.isRetriable());
     }
 
+    @Test
+    public void testSyncGroupRequestWithFencedInstanceIdException() {
+        setupCoordinator();
+        mockClient.prepareResponse(groupCoordinatorResponse(node, Errors.NONE));
+
+        final String memberId = "memberId";
+        final int generation = -1;
+
+        mockClient.prepareResponse(joinGroupFollowerResponse(generation, memberId, JoinGroupResponse.UNKNOWN_MEMBER_ID, Errors.NONE));
+        mockClient.prepareResponse(syncGroupResponse(Errors.FENCED_INSTANCE_ID));
+        try {
+            coordinator.ensureActiveGroup();
+            fail("Failed to catch expected FencedInstanceIdException");
+        } catch (FencedInstanceIdException e) {
```

Comment:
nit: I'd suggest a better catch which is just define it generally, and then assert its actual instanceof. I.e.:

```
catch (RuntimeException e) {
     assertTrue(e instanceof FencedInstanceIdException, "some error message");
}
```

That way, we are not strictly depend on the error message, and we can have a meaningful error message if some bugs cause it to throw a different exception. Ditto elsewhere.

Reply:
Thank you for the suggestion!